app/models/base_organisation.rb

```diff
@@ -54,5 +57,32 @@ def setup_categories
   def has_relation?(category)
     self.category_organisations.any? {|v| v.category_id == category.id}
   end
+  
+  def slug_candidates
+    [
+      :short_name,
+      :prolonged_name,
+      :orged,
+      :name
+    ]
+  end
+  
+  def short_name
+    slug_words.first(3).join('-')
+  end
+  
+  def prolonged_name
+    "#{self.short_name}-#{slug_words[-2]}-#{slug_words[-1]}"
+  end
+  
+  def orged
+    "#{slug_words.join('-')}-org"
+  end
+  
+  def slug_words
+    not_wanted = ['the', 'of', 'for', 'and', 'in', 'to']
+    words = "#{self.name}".delete('\'','-').scan(/\b\w+\b/).map(&:downcase)
+    words.reject {|w| not_wanted.include?(w) }
+  end
```

Comment:
do you want to have a go?  only if you're interested ... there's more constants that could be pulled out.

I guess I'm just a little nervous about adding lots more public interface to our already large base_org (which has a knock on effect on org and proposed_org).  I'm trying to follow the SOLID: single responsibility principle:

https://robots.thoughtbot.com/back-to-basics-solid